contracts/oracles/ChainlinkOracle.sol

```diff
@@ -31,7 +34,9 @@ contract ChainlinkOracle is IOracle {
     function getRate() external view override returns (uint256) {
         uint256 feedDecimals = uint256(dataFeed.decimals());
 
-        (, int256 rate, , , ) = dataFeed.latestRoundData();
+        (, int256 rate, , uint256 updatedAt, ) = dataFeed.latestRoundData();
+        
+        require(updatedAt >= block.timestamp - tolerance, "ChainlinkOracle#getRate: STALE_RATE");
```

Comment:
Do we need to use safeMath here or we are ok? Can you add a comment about why we are not using safe match? maybe we can just check that `tolerance` is lower than `block.timestamp`